python/hail/methods/tests.py

```diff
@@ -65,6 +69,122 @@ def test_sample_qc(self):
+            ids = []
+            with hadoop_read(path) as f:
+                for l in f:
+                    r = l.strip().split('\t')
+                    assert(len(r) == 2)
+                    ids.append(r[1])
+            return ids
+
+        def load_rel(ns, path):
+            rel = np.zeros((ns, ns))
+            with hadoop_read(path) as f:
+                for i,l in enumerate(f):
+                    for j,n in enumerate(map(float, l.strip().split('\t'))):
+                        rel[i,j] = n
+                    assert(j == i)
+                assert(i == ns - 1)
+            return rel
+
+        def load_grm(ns, nv, path):
+            m = np.zeros((ns, ns))
+            with utils.hadoop_read(path) as f:
+                i = 0
+                for l in f:
+                    row = l.strip().split('\t')
+                    assert(int(row[2]) == nv)
+                    m[int(row[0])-1, int(row[1])-1] = float(row[3])
+                    i += 1
+
+                assert(i == ns * (ns + 1) / 2)
+            return m
+
+        def load_bin(ns, path):
+            m = np.zeros((ns, ns))
+            with utils.hadoop_read_binary(path) as f:
+                for i in range(ns):
+                    for j in range(i + 1):
+                        b = f.read(4)
+                        assert(len(b) == 4)
+                        m[i, j] = unpack('<f', bytearray(b))[0]
+                left = f.read()
+                print(left)
+                assert(len(left) == 0)
+            return m
+
+        b_file = utils.new_temp_file(prefix="plink")
+        rel_file = utils.new_temp_file(prefix="test", suffix="rel")
+        rel_id_file = utils.new_temp_file(prefix="test", suffix="rel.id")
+        grm_file = utils.new_temp_file(prefix="test", suffix="grm")
+        grm_bin_file = utils.new_temp_file(prefix="test", suffix="grm.bin")
+        grm_nbin_file = utils.new_temp_file(prefix="test", suffix="grm.N.bin")
+
+        dataset = self.get_dataset()
+        n_samples = dataset.count_cols()
+        dataset = dataset.annotate_rows(AC=agg.sum(dataset.GT.num_alt_alleles()),
+                              n_called=agg.count_where(functions.is_defined(dataset.GT)))
+        dataset = dataset.filter_rows((dataset.AC > 0) & (dataset.AC < 2 * dataset.n_called))
+        dataset = dataset.filter_rows(dataset.n_called == n_samples).persist()
+
+        dataset.to_hail1().export_plink(b_file)
+
+        sample_ids = [row.s for row in dataset.cols_table().select('s').collect()]
+        n_variants = dataset.count_rows()
+        assert(n_variants > 0)
+
+        grm = methods.grm(dataset)
+        grm.export_id_file(rel_id_file)
+
+        ############
+        ### rel
+
+        p_file = utils.new_temp_file(prefix="plink")
+        syscall('''plink --bfile {} --make-rel --out {}'''
+                .format(utils.get_URI(b_file), utils.get_URI(p_file)), shell=True)
+        assert(load_id_file(p_file + ".rel.id") == sample_ids)
```

Comment:
a bunch of the asserts here should be `self.assertEqual(x, y)` to follow the unittest model